tools/gyp/pylib/gyp/generator/msvs.py

```diff
@@ -3633,7 +3633,10 @@ def _AddSources2(
                     elif file_name.startswith("$("):
                         file_name = re.sub(r"^\$\([^)]+\)\\", "", file_name)
                     detail.append(["ObjectFileName", "$(IntDir)\\" + file_name])
-                grouped_sources[group].append([element, {"Include": source}] + detail)
+                element_node = [element, {"Include": source}]
+                if element == "MARMASM":
+                    element_node.append(["PreprocessedFileName", source + ".pp"])
```

Comment:
This looks vaguely similar to https://github.com/nodejs/gyp-next/pull/162.

(gyp-next is where this change should go but maybe you can coordinate with the author of that pull request.)

Reply:
This code must avoid generating files in the source tree. The generated files must go to the $(IntDir). PR #46228 solves this issue without changing GYP by adding `directory.build.props` file to override `PreprocessedFileName` locally.